app/client/components/Clickr.js

```diff
@@ -0,0 +1,96 @@
+
+import React from 'react';
+
+export default class Clickr extends React.Component {
+
+    state = {
+        count: 0
+    };
+
+    handleClickPlus1 = (e) => {
+      var { count } = this.state;
+      var nextCount = count + 1;
+
+      if (nextCount <= 20) {
+        this.setState({
+          count: nextCount,
+        });
+      }
+    }
+
+    handleClickPlus5 = (e) => {
+      var { count } = this.state;
+      var nextCount = count + 5;
+
+      if (nextCount > 20) {
+        this.setState({
+          count: 20,
+        });
+      } else {
+        this.setState ({
+          count: nextCount,
+        });
+      }
+    }
+
+    handleClickminus1 = (e) => {
+      var { count } = this.state;
+      var nextCount = count - 1;
+
+      if (nextCount >= -20) {
+        this.setState({
+          count: nextCount,
+        });
+      }
+    }
+
+    handleClickminus5 = (e) => {
+      var { count } = this.state;
+      var nextCount = count - 5;
+
+      if (nextCount < -20) {
+        this.setState({
+          count: -20,
+        });
+      } else {
+        this.setState ({
+          count: nextCount,
+        });
+      }
+    }
+
+    render() {
+        var { count } = this.state;
+
+        var buttonStyle = {
+          margin: '10',
+          display: 'inline-block',
+        };
+
+        var counterStyle = {
+          backgroundColor: 'black',
+          display: 'inline-block',
+          width: '30',
+          height: '20',
+          textAlign: 'center',
+        };
+
+        if (count > 0) {
+          counterStyle.backgroundColor = 'green';
+        } else if (count < 0) {
+          counterStyle.backgroundColor = 'red'
+        }
+
+        return (
+            <div>
+              <button style={buttonStyle} onClick={this.handleClickminus5}>-5</button>
```

Comment:
This is not a bug or a bad thing but is more a refactoring that needs to introduce the concept of a "thunk".

> A thunk is a function that returns a function.

```
// this is the thunk
function sumTo(p1) {
  return function(p2) {
    return p1 + p2;
  }
}

// so you can create many functions now:
var sumTo5 = sumTo(5);
var sumTo1 = sumTo(1);

// and you use the latest functions:
sumTo5(2) === 7
````

In this exercise you have a `handleClick...` callback and you have implemented many of those (-5, -1, +1, +5).
This can me simplified with thunks:

```
// this is how you dinamically generate callbacks from a thunk:
var { handleClick } = this;
<button ... onClick={handleClick(1)} />
<button ... onClick={handleClick(5)} />
<button ... onClick={handleClick(-1)} />
<button ... onClick={handleClick(-5)} />
```

the thunk is very close to what you already created but incapsulate a bit more logic:

```
// thunk as arrow function in the component:
handleClick = increment => {
  // this is the real callback
  return e => {
    var { count } = this.state;

    // compute the next value based on the increment that was set when
    // the callback was created:
    var nextCount = count + increment;
    
    // decide whether to modify the state
    if (nextCount >= -20 && count <= 20) {
      this.setState({ count: nextCount });
    } 
  }
}
```

NOTE: this implementation was written by heart and I have no idea if it's working or not. You should do this refactoring as new exericise (I'll add it to Trello)

NOTE: this thing require you to study `variable's scope` in Javascript and to understand it very very very very (very) well. It is a cornerstone for good Javascript coding :-)